src/olympia/lib/crypto/signing.py

```diff
@@ -60,15 +62,53 @@ def use_promoted_signer(file_obj, promo_group):
     )
 
 
+def add_guid(file_obj):
+    with storage.open(file_obj.current_file_path) as fobj:
+        # Get the file data and add the guid to the manifest if waffle switch is enabled
+        if waffle.switch_is_active('add-guid-to-manifest'):
+            with zipfile.ZipFile(fobj, mode='r') as existing_zip:
+                manifest_json = json.loads(existing_zip.read('manifest.json'))
```

Comment:
What happens if `manifest.json` does not exist in the `existing_zip`? Or are we guaranteed that this code will never run if that is the case?

Reply:
It wouldn't have gotten through the linter and other validation with a `manifest.json`, and if it did then I think we're okay erroring - autograph isn't going to sign it anyway.